micro-rdk/src/common/registry.rs

```diff
@@ -212,138 +230,150 @@ impl ComponentRegistry {
         }
     }
     #[cfg(feature = "camera")]
-    pub fn register_camera(
+    pub fn register_camera<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
```

Comment:
BTW, I realized you can also write these in the equivalent form:

`pub fn register_x(&mut self, model: impl Into<String>)`, if you like that better.